lib/pbench/server/hierarchy.py

```diff
@@ -228,118 +207,124 @@ def __init__(self, name, path, config):
             self.INCONCEIVABLE: DictOfList("Inconceivable Conditions:"),
         }
 
-    def add_tarballs(self, controller):
+    def add_tarballs(self, controller: str):
+        """Add controllers which consist of tarballs"""
         self.tarballs.append(controller)
 
-    def dump(self, fp):
+    def dump(self, fp: IO) -> int:
         """Checks and Output collected data"""
         cnt = 0
         if self.bad_controllers:
             self.output_bad_controllers(fp)
-            cnt += 1
+            cnt = 1
         for controller in sorted(self.controllers):
-            check = self.check_controller_in_list(controller)
+            check = self.is_controller_in_list(controller)
             if check or controller not in self.tarballs:
```

Comment:
I don't think you've got the names quite right, here:  when would you ever expect to find a "controller" in "tarballs"?  😃